kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/handlers/TopicListHandler.java

```diff
@@ -1,41 +1,57 @@
-import java.util.HashMap;
+import io.vertx.ext.web.handler.graphql.VertxDataFetcher;
+import io.vertx.ext.web.impl.RoutingContextWrapper;
+
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
+import java.util.Set;
 import java.util.function.Predicate;
-import java.util.stream.Collectors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class TopicListHandler {
 
-    public static final Map<String, Topic> SAMPLE_TOPICS = new HashMap<>();
-    static {
-        SAMPLE_TOPICS.put("topic1", Topic.create("topic1", false, 3, 3));
-        SAMPLE_TOPICS.put("topic2", Topic.create("topic2", false, 1, 1));
-        SAMPLE_TOPICS.put("topic2-a", Topic.create("topic2-a", false, 1, 1));
-        SAMPLE_TOPICS.put("topic2-b", Topic.create("topic2-b", false, 1, 1));
-        SAMPLE_TOPICS.put("topic2-c", Topic.create("topic2-c", false, 1, 1));
-        SAMPLE_TOPICS.put("topic3", Topic.create("topic3", false, 6, 1));
-    }
+    public static VertxDataFetcher topicListFetch(AdminClientProvider acp) {
+        VertxDataFetcher<Types.TopicList> dataFetcher = new VertxDataFetcher<>((env, prom) -> {
+            RoutingContextWrapper routingContext = env.getContext();
+            routingContext.request().headers().get("token");
 
-    public static void getTopicList(final DataFetchingEnvironment env, final Promise<List<Topic>> promise) {
-        promise.complete(
-            SAMPLE_TOPICS
-                .values()
-                .stream()
-                .filter(byTopicName(env.getArgument("filter")))
-                .collect(Collectors.toList()));
+            Promise<Set<String>> describeTopicsNamesPromise = Promise.promise();
+            acp.listTopics(describeTopicsNamesPromise);
+            describeTopicsNamesPromise.future()
+                    .onComplete(topics -> {
+                        Types.TopicList topicList = new Types.TopicList();
+                        List<Types.Topic> items = new ArrayList<>();
+                        topics.result().forEach(top -> {
+                            if (byTopicName(env.getArgument("search")).test(top)) {
+                                Types.Topic ton = new Types.Topic();
+                                ton.setName(top);
+                                items.add(ton);
+                            }
+                        });
+                        topicList.setItems(items);
+                        prom.complete(topicList);
+                    });
+        });
+        return dataFetcher;
     }
 
-    private static Predicate<Topic> byTopicName(String filterParameter) {
+    private static Predicate<String> byTopicName(String regexParam) {
         return topic -> {
-            if (filterParameter == null) {
+            if (regexParam == null) {
                 return true;
             } else {
-                return topic.getName().contains(filterParameter);
+                Pattern pattern = Pattern.compile(regexParam);
```

Comment:
I'd suggest we avoid a regex pattern being pattern of the public API to avoid having to guard for:
https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS

Reply:
what do you suggest for users who want to search for more topics matching a pattern?

Reply:
To me it is really a public API design question.   What function do we anticipate the user needing?  Do we have established practices in other APIs that we should follow?

IMHO I suspect a % wildcard would be sufficient to most user needs and is familiar to most from a technical domain.